validphys2/src/validphys/commondataparser.py

```diff
@@ -40,22 +108,825 @@
+        folder_name = <experiment>_<process>_<energy>{_<extra>}
+    but note that this convention is still not fully defined.
+
+    This function returns a commondata object constructed by parsing the metadata.
+
+    Once a variant is selected, it can no longer be changed
+
+    Note that this function reproduces `parse_commondata` below, which parses the
+    _old_ file format
+    """
+    # Now parse the data
+    data_df = metadata.load_data_central()
+    # the uncertainties
+    uncertainties_df = metadata.load_uncertainties()
+    # and the kinematics
+    kin_df = metadata.load_kinematics()
+
+    # Once we have loaded all uncertainty files, let's check how many sys we have
+    nsys = len(
+        [i for i in uncertainties_df.columns.get_level_values(0) if not i.startswith("stat")]
+    )
+
+    # Backwards-compatibility
+    # Finally, create the commondata by merging the dataframes in the old commondata_table
+
+    procname = metadata.process_type  # nnpdf_metadata["nnpdf31_process"]
+    kin_df = kin_df[metadata.kinematic_coverage]
+    kin_df.columns = KIN_NAMES
+    kin_df["process"] = procname
+
+    kin_df = kin_df[["process"] + KIN_NAMES]
+
+    # For the uncertainties, create a simplified version to concatenate
+    # and save the systype information
+    new_columns = []
+    systypes = {"treatment": [], "name": []}
+    for col in uncertainties_df.columns:
+        if col[0].startswith("stat"):
+            new_columns.append("stat")
+        else:
+            # if it is syst add the ADD/MULT information
+            new_columns.append(col[1])
+            systypes["treatment"].append(col[1])
+            systypes["name"].append(col[2])
+
+    uncertainties_df.columns = new_columns
+
+    commondata_table = pd.concat([kin_df, data_df, uncertainties_df], axis=1)
+    systype_table = pd.DataFrame(systypes, index=range(1, nsys + 1))
+    systype_table.index.name = "sys_index"
+
+    # TODO: Legacy compatibility
+    # 1. Add a stat column if it doesn't exist
+    # 2. Transform multiplicative uncertainties into % as it was done in the older version
+
+    if "stat" not in commondata_table:
+        commondata_table["stat"] = 0.0
+
+    if "MULT" in commondata_table:
+        commondata_table["MULT"] = commondata_table["MULT"].multiply(
+            100 / commondata_table["data"], axis="index"
+        )
+
+    # TODO: For the time being, fill `legacy_name` with the new name if not found
+    legacy_name = metadata.name
+
+    if (old_name := new_to_legacy_map(metadata.name, metadata.applied_variant)) is not None:
+        legacy_name = old_name
+
+    return CommonData(
+        setname=metadata.name,
+        ndata=metadata.ndata,
+        commondataproc=procname,
+        nkin=3,
+        nsys=nsys,
+        commondata_table=commondata_table,
+        systype_table=systype_table,
+        legacy=False,
+        legacy_name=legacy_name,
+        kin_variables=metadata.kinematic_coverage,
+    )
```

Comment:
What about the `systematics_table` argument? In the dataclass definition, that is not Optional.

Reply:
It is autogenerated during the `__post_init__` of that class.

Reply:
Right, but then it should set as Optional (?)

Reply:
tbf it does have a default value set, but I can mark it explicitly